src/raqm.c

```diff
@@ -854,6 +854,29 @@ raqm_add_font_feature (raqm_t     *rq,
     if (!new_features)
       return false;
 
+    if (fea.start != HB_FEATURE_GLOBAL_START)
+    {
+      if (rq->text_utf8)
+      {
+        fea.start = _raqm_u8_to_u32_index (rq, fea.start);
```

Comment:
We probably need a helper function or a macro that encapsulates this logic, something like `_rq_encoding_to_u32_index()` that would check the encoding and do the mapping if necessary.